### PR TITLE
Stripe Free Trial on Subscription

### PR DESCRIPTION
**Pull request by husobee**
### Summary

updates a free trial for talk.brave.com in stripe checkout session

### Type of change ( select one )

- [x] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: payment/controllers.go [hunk 1]
```diff
@@ -286,6 +287,47 @@ func CreateOrder(service *Service) handlers.AppHandler {
 	})
 }
 
+// SetOrderTrialDaysInput - SetOrderTrialDays handler input
+type SetOrderTrialDaysInput struct {
+	TrialDays int64 `json:"trialDays" valid:"int"`
+}
+
+// SetOrderTrialDays is the handler for cancelling an order
+func SetOrderTrialDays(service *Service) handlers.AppHandler {
```
Comment (evq) [inline payment/controllers.go:296]: I think you need a call to `ValidateOrderMerchantAndCaveats`

File: payment/service.go [hunk 2]
```diff
@@ -312,6 +316,47 @@ func (s *Service) CancelOrder(orderID uuid.UUID) error {
 	return s.Datastore.UpdateOrder(orderID, OrderStatusCanceled)
 }
 
+// SetOrderTrialDays set the order's free trial days
+func (s *Service) SetOrderTrialDays(ctx context.Context, orderID *uuid.UUID, days int64) error {
+	// get the order
+	err := s.Datastore.SetOrderTrialDays(ctx, orderID, days)
```
Comment (evq) [inline payment/service.go:322]: might be simpler to just have this return updated order?

**Review by evq — CHANGES_REQUESTED** [review]

File: payment/controllers.go (hunk 1 above)
Comment (husobee) [reply payment/controllers.go:296]: addressed: https://github.com/brave-intl/bat-go/pull/945/commits/bc0dfc7a96b474900bad381e9553bed06c6491a1

File: payment/service.go (hunk 2 above)
Comment (husobee) [reply payment/service.go]: addressed: https://github.com/brave-intl/bat-go/pull/945/commits/bc0dfc7a96b474900bad381e9553bed06c6491a1

**Review by evq — APPROVED** [review]

**Review by evq — APPROVED** [review]

**Pull request merged**